### ISSUE TITLE
Interface buttons to pick up specific creature types

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1209]
To pickup units and assign them to a fight you can only select the 'select a fighter in hand button' to get a random (?) fighter unit. This system is too basic.

The multiplayer game I played yesterday was the first time playing the hero faction, and I had no idea which creatures were of which type or which tier. In fact, without reading the manual there's no way to understand the combat system at all, and the GUI does nothing to support the combat system at all.

What would be a major help is if instead of a single 'fighter' button you have three buttons, one each for psychical/magical/elemental creatures. This allows the player to immediately understand the 3 creature types and more easily understand the rock-paper-scissors relation between them.
Furthermore for experienced players it allows you when you see a rival creature to immediately select creatures of the proper type to defeat it.

To make these buttons especially user friendly:
- Make the creature type triangle visible on the buttons. (That magic beats psychical, etc.)
- Show the amount of creatures of that type on the button.
- Allow Ctrl+Click to pickup creatures of the highest tier first, and Shift+Click to pick lowest tier creatures first.

**Comment by hwoarangmy** [comment]
ATM, the pick fighter button is pretty messy and will try to pickup the highest level creature available as well as the creatures fleeing. I had in mind to had another button next to the pick fighter button so that one would allow to pickup the most ready to fight creature and the other one the most in need of rest creature.

Concerning your idea of using one button per creature type, note that a creature might be able to deal several types of damage at once. But for this kind of creature, I guess that pulling them through several buttons is acceptable.

@OpenDungeons/developers WDYT ?

**Comment by Loobinex** [comment]
Can creatures also be vulnerable to multiple types of damage? If not what they are strong against defensively can determine their type.

**Comment by hwoarangmy** [comment]
yes, they can. Basically, they have a default defense/attack value per type (value at level 1) and a bonus per type per level. So, the problem is the same ^^

**Comment by Danimal696** [comment]
i think its more sensible to create three buttoms, for melee, ranged and mages/support. I guess thats the best classification for now, since melee will always have more hp and the others need to be dropped at some distance from the fight; that is true regardless of what damage type they deal or resist.

**Comment by Loobinex** [comment]
Danimal696: I'm afraid to say If that's really is true you'd be much better of dropping the damage types completely and base the creature types around Melee/Ranged/Support.

**Comment by Danimal696** [comment]
Then what do you suggest? its not the same to drop a cultist or a kreatur when you press magic damagers, the same as an orc or an elf. same kind of damage , but different roles

**Comment by Loobinex** [comment]
I'm suggesting to simplify or overhaul your creature types. Right now you've got lots of complexity without any real strategic choices. That's not system that delivers fun.

Currently you have Tiers (1,2,3,4) x Types (Physical, Magical, Elemental) x Roles (Melee, Ranged, Support), so already 36 combinations without taking into account that one creature can also have multiple types, multiple roles and this can change on level-up when new skills are acquired.
Now I have to admit I haven't really looked into how I as a player can use all this to my advantage, but my impression is your focus is simply getting as much high tier creatures as possible spread out over the types and either drop your blob on the enemy blob or when possible play rock-paper-scissors. What I'm trying to say is that the complexity of the system does not lead to lots of gameplay choices and complex decision making. Meaning you got it in reverse.

And how does this relate to the GUI buttons? Well, my experience is when you can't design a good interface/gui for a system, it's usually not a very system to begin with. Why don't you make a simple drawing of your dream interface for creature/combat management in this game, that supports all the strategic choices you want to make. You might find you'll end up with more buttons than creatures and that some general rethinking is in order. And perhaps you do manage in which case share the GUI design, but I'm pretty sure you'll come to a better understanding of how your system works in practice.

Have you for example thought about not having the types, but increasing the number of roles? (Area attack, debuffer, glass cannon, whatever). This shifts the focus from an underlying invisible mechanic towards the player actually getting to know the creatures (e.g. "Wow that enemy Golem sure hits hard, do I still have a creature that can freeze him? No, well better use ranged creatures then that can retreat backwards".)
